docs/ExtendingHeartbeatProperties.md

```diff
@@ -0,0 +1,96 @@
+_This document was last updated 11/25/2016 and is applicable to SDK version 2.5.0-beta2._
+
+# Extending Heartbeat Properties in Application Insights #
+
+The .NET Application Insights SDKs provide a new feature called Health Heartbeat. This feature
+sends health related data as well as environment-specific information at pre-configured
+intervals. The feature will allow you to extend the properties that will be sent every interval,
+and will also allow you to set a flag denoting a healthy or unhealthy status for each property
+you add to the heartbeat payload.
+
+## A General Code Example ##
+
+In order to add the extended properties of your choice to the Health Heartbeat as a developer
+of an ITelemetryModule, you can follow the following pattern. Note that you must first add the
```

Comment:
Is it only applicable to developer of ITelemetryModule?

Reply:
For now, that is the idea yes.

Reply:
whats preventing someone else from using it? Any code can get a hold of IHeartbeatPropertyManager, and add/set HB right?

Reply:
Nothing is preventing anyone from using it, that is simply the focus here. Perhaps my wording could be improved here but for the currently intended audience I believe it's good. Once we make this publicly available (and indicate it as such via documentation) I believe the very-easily-accessible API we provide at that time will supersede this file altogether.